test/go-tests/test_gitcommitid.go

```diff
@@ -0,0 +1,255 @@
+	_, err = ExecuteCommand(fmt.Sprintf("kubectl create configmap -n %s lighthouse-config-%s --from-literal=sli-provider=my-sli-provider", GetKeptnNameSpaceFromEnv(), projectName))
+	require.Nil(t, err)
+
+	var evaluationFinishedEvent *models.KeptnContextExtendedCE
+	evaluationFinishedPayload := &keptnv2.EvaluationFinishedEventData{}
+
+	//first part
+
+	t.Log("storing SLO file")
+	commitID := storeWithCommit(t, projectName, "hardening", serviceName, gitCommitIDSLO, "slo.yaml")
+	t.Logf("commitID is %s", commitID)
+
+	t.Log("triggering the evaluation")
+	_, evaluationFinishedEvent = performResourceServiceEvaluationTest(t, projectName, serviceName, commitID)
+
+	t.Log("checking the finished event")
+	err = keptnv2.Decode(evaluationFinishedEvent.Data, evaluationFinishedPayload)
+	require.Nil(t, err)
+
+	require.Equal(t, evaluationFinishedPayload.EventData.Result, keptnv2.ResultType("pass"))
+	require.Equal(t, evaluationFinishedPayload.EventData.Status, keptnv2.StatusType("succeeded"))
+	t.Log("evaluation is finished and correct")
+
+	//second part
+
+	t.Log("storing second invalid SLO file")
+	commitID1 := storeWithCommit(t, projectName, "hardening", serviceName, "gitCommitIDSLO", "slo.yaml")
+	t.Logf("new commitID is %s", commitID1)
+
+	t.Log("triggering the evaluation again with the old commitID")
+	_, evaluationFinishedEvent = performResourceServiceEvaluationTest(t, projectName, serviceName, commitID)
+
+	t.Log("checking the finished event again with the old commitID")
+	err = keptnv2.Decode(evaluationFinishedEvent.Data, evaluationFinishedPayload)
+	require.Nil(t, err)
+
+	require.Equal(t, evaluationFinishedPayload.EventData.Result, keptnv2.ResultType("pass"))
+	require.Equal(t, evaluationFinishedPayload.EventData.Status, keptnv2.StatusType("succeeded"))
+	t.Log("evaluation is finished and correct")
+}
+
+func performResourceServiceEvaluationTest(t *testing.T, projectName string, serviceName string, commitID string) (string, *models.KeptnContextExtendedCE) {
+	keptnContext := ""
+	source := "golang-test"
+
+	t.Log("sent hardening.evaluation.triggered with commitid: ", commitID)
+	_, err := ExecuteCommand(fmt.Sprintf("keptn trigger evaluation --project=%s --stage=hardening --service=%s --start=2022-01-26T10:05:53.931Z --end=2022-01-26T10:10:53.931Z --git-commit-id=%s", projectName, serviceName, commitID))
+	require.Nil(t, err)
+
+	var getEvaluationTriggeredEvent *models.KeptnContextExtendedCE
+	require.Eventually(t, func() bool {
+		t.Log("checking if hardening.evaluation.triggered event is available")
+		event, err := GetLatestEventOfType(keptnContext, projectName, "hardening", keptnv2.GetTriggeredEventType("hardening."+keptnv2.EvaluationTaskName))
+		if err != nil || event == nil {
+			return false
+		}
+		getEvaluationTriggeredEvent = event
+		return true
+	}, 1*time.Minute, 10*time.Second)
+
+	t.Log("got hardening.evaluation.triggered event, checking commitid")
+	require.Equal(t, commitID, getEvaluationTriggeredEvent.GitCommitID)
```

Comment:
I think this fails to compile, since the version of `go-utils` here is using `GitCommitid` pls update go-utils

Reply:
here is the update for go-utils, which was not merged yet: https://github.com/keptn/go-utils/pull/401

Reply:
you can update go-utils here to `8c0f3dea9473493293bf4355c5bb99ae0b94c1ad` (latest commit in PR for changes in go-utils) to have it compiled, later I think a github action should take care of updating the version to master automatically.

qq: this test can only be executed on GH once the other PRs got merged right?

Reply:
I updated that and tried locally and it worked, so I reverted it, to habe it mergable to master :)

This test will be runnable only when using resource-service

Reply:
> I updated that and tried locally and it worked, so I reverted it, to habe it mergable to master :)
Sorry, I don't get it. By "reverting" it it's not able to compile anymore, i.e. when merging this to master you would get a compile error (Until the Github Action comes by and updates go-utils to current master)

Reply:
After merging the go-utils update and also the dependency update https://github.com/keptn/keptn/commit/4aad431293a732f18a271b4fb8f4485ea70d0627 everything should be buildable and working now

Reply:
thanks, yes. I just brought this up as a hint for the future, that you usually don't need to take care of it - just do your PR with a link to the commit in go-utils that contain the required changes.

Reply:
thanks :) all green, approve ? :)